### PR TITLE
Feature photos for bought album

### PR DESCRIPTION
**Pull request by christiangelone**
## Summary
  * Covers that album service retrives photos for a given id
  * Covers that a not logged user cannot ask for photos of its bought album with given id
  * Covers that logged user or admin can ask for photos of its bought album with given id
  * Covers that a logged user or admin cannot ask for photos of a not bought album with given id

## Known Issues
 * No Issues

## Trello Card
  * https://trello.com/c/6TsqOGCi

### CODE REVIEW

File: app/controllers/users.js [hunk 1]
```diff
@@ -143,10 +143,44 @@ exports.boughtAlbums = (req, res, next) => {
...
+        next(errors.databaseError(err));
+      });
   } else {
     next(errors.forbiddenError(new Error('You cannot see others albums')));
   }
 };
+
+exports.photosBoughtAlbum = (req, res, next) => {
+  const albumId = parseInt(req.params.id);
+  AlbumPurchase.findOne({
```
Comment (gonziesc) [inline app/controllers/users.js]: return

File: app/controllers/users.js [hunk 2]
```diff
@@ -143,10 +143,44 @@ exports.boughtAlbums = (req, res, next) => {
...
+    where: {
+      userId: req.user.id,
+      albumId
+    }
+  })
+    .then(purchase => {
+      if (purchase) {
+        AlbumService.getPhotosForAlbumWithId(purchase.albumId)
```
Comment (gonziesc) [inline app/controllers/users.js]: return

File: app/services/albums.js [hunk 3]
```diff
@@ -27,10 +27,40 @@ exports.ALBUMS_SAMPLE = [
 
 exports.ALBUM_SAMPLE = id => exports.ALBUMS_SAMPLE[id - 1];
 
+exports.PHOTOS_FOR_ALBUM_SAMPLE = id => {
```
Comment (gonziesc) [inline app/services/albums.js]: why the sample is in the service?

**Review by gonziesc — CHANGES_REQUESTED** [review]

File: app/services/albums.js (hunk 3 above)
Comment (christiangelone) [reply app/services/albums.js]: because at first was handy, but you are right i'll put it somewhere else

File: app/controllers/users.js (hunk 2 above)
Comment (gonziesc) [reply app/controllers/users.js]: what abpout this?

**Review by gonziesc — CHANGES_REQUESTED** [review]

File: app/services/albums.js [hunk 4]
```diff
@@ -2,30 +2,7 @@ const axios = require('axios');
...
-  },
-  {
-    userId: 2,
-    id: 4,
-    title: 'non esse culpa molestiae omnis sed optio'
-  }
-];
-
-exports.ALBUM_SAMPLE = id => exports.ALBUMS_SAMPLE[id - 1];
+exports.SAMPLES = require('./samples/album_samples');
```
Comment (gonziesc) [inline app/services/albums.js:5]: why dont you import it where you need it?

File: app/services/samples/album_samples.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    title: 'omnis laborum odio'
+  },
+  {
+    userId: 2,
+    id: 4,
+    title: 'non esse culpa molestiae omnis sed optio'
+  }
+];
+
+exports.ALBUM_SAMPLE = id => exports.ALBUMS_SAMPLE[id - 1];
```
Comment (gonziesc) [inline app/services/samples/album_samples.js:24]: does this work?

**Review by gonziesc — CHANGES_REQUESTED** [review]

File: app/services/samples/album_samples.js (hunk 5 above)
Comment (christiangelone) [reply app/services/samples/album_samples.js:24]: yes

File: app/services/albums.js (hunk 4 above)
Comment (christiangelone) [reply app/services/albums.js:5]: its handy
at the same time you know that those samples corresponds to that service
plus you dont have requires everywhere

File: app/services/samples/album_samples.js (hunk 5 above)
Comment (gonziesc) [reply app/services/samples/album_samples.js:24]: oh sorry, i misunderstood this

**Review by gonziesc — APPROVED** [review]

**Pull request merged**